macaroonbakery/httpbakery/client.py

```diff
@@ -290,11 +290,20 @@ def hook(response, *args, **kwargs):
     return hook
 
 
-def extract_macaroons(headers):
+def extract_macaroons(headers_or_request):
     ''' Returns an array of any macaroons found in the given slice of cookies.
-    @param headers: dict of headers
+    If headers implements a get_header method, that will be used
+    instead of get to retrieve headers.
+    @param headers_or_request: dict of headers or a
+    urllib.request.Request-like object.
     @return: An array of array of mpy macaroons
     '''
+    def get_header(key, default=None):
```

Comment:
by putting these here they'll get reprocessed on function invocations. I think these would just be private methods outside of this
    
    def _get_header(headers_or_request, key, default=None)

Same with the add_macaroon to _add_macaroon

Reply:
Discussed online; leaving as is for now.